cmd/snap/cmd_info.go

```diff
@@ -382,13 +382,13 @@ func (iw *infoWriter) maybePrintBuildDate() {
 }
 
 func (iw *infoWriter) maybePrintLinks() {
+	if len(iw.theSnap.Links) == 0 {
+		return
+	}
 	contact := strings.TrimPrefix(iw.theSnap.Contact, "mailto:")
 	if contact != "" {
 		fmt.Fprintf(iw, "contact:\t%s\n", contact)
 	}
-	if !iw.verbose || len(iw.theSnap.Links) == 0 {
```

Comment:
One obvious downside is that links contain all of the links provided in the metadata, including the ones that are duplicated by other fields shown by default (eg. contact). See:

```
$ ./test-snap info ohmygiraffe
name:      ohmygiraffe
summary:   oh my giraffe
publisher: James Tigert (kz6fittycent)
store-url: https://snapcraft.io/ohmygiraffe
contact:   https://github.com/kz6fittycent/ohmygiraffe
links:
  contact:
    - https://github.com/kz6fittycent/ohmygiraffe
  website:
    - https://www.ohmygiraffe.com/
```

the same entry is duplicated in `contact`, and `links.contact` and this would affect every snap you try this for. IMO less than ideal UX.

Here is another one:
```
$ ./test-snap info vault  
name:      vault
summary:   Vault is a tool for securely accessing secrets.
publisher: Canonical✓
store-url: https://snapcraft.io/vault
links:
  issues:
    - https://github.com/canonical/snap-vault/issues
  source:
    - https://github.com/hashicorp/vault
  website:
    - https://github.com/canonical/snap-vault
    - https://www.vaultproject.io/
```

What if instead of displaying all links by default, `snap info` was to show some well known attributes as top level entries? Eg. `links.source.[0]` could be displayed as `source-code: ...`, and `links.website[0]` as the value of `website: ..`, but still keep all of the metadata behind `--verbose` flag.

The store API mentions `links` https://api.snapcraft.io/docs/info.html but there is no information what to expect there. However, the store page seems to associate specific meaning with links, eg https://snapcraft.io/vault

Reply:
I'd had similar thoughts, my initial concern that was if there's any scripts in the wild that do e.g `snap info $snap | grep "contact: " or similar you could break some assumptions in the regex. The `contact:` effectively appearing twice is the existing behavior when `--verbose` is passed today and so contact being visible twice isn't a regression from the perspective of `--verbose` being passed.

If we were fine for changing the formatting of the output more, I'd personally propose the output would have been:
```
Contacts: 
  - Example 1
  - Example 2
Source-Code:
  - Example 1
  - Example 2
 etc:
```

Using the `Vault` snap as an example, the current output without `verbose` in `2.63` is:
```
ames@James-Desktop:~$ snap info vault
name:      vault
summary:   Vault is a tool for securely accessing secrets.
publisher: Canonical✓
store-url: https://snapcraft.io/vault
license:   BUSL-1.1
description: |
  A modern system requires access to a multitude of secrets: database
  credentials, API keys for external services, credentials for
  service-oriented
  architecture communication, etc. Understanding who is accessing what
  secrets
  is already very difficult and platform-specific. Adding on key rolling,
  secure storage, and detailed audit logs is almost impossible without a
  custom
  solution. This is where Vault steps in.
  
  This snap is maintained by Canonical.
  
  **Usage**
  
  * To start the Vault service, edit the configuration file at
  `/var/snap/vault/common/vault.hcl` and start the service with `sudo snap
  start vault.vaultd`.
  
  * To use the Vault Client, run `vault` commands. For example: `vault
  status`.
snap-id: bIb4p4yWWjyZdo2EU64whkZhw9QYYsMH
channels:
```

So for that exact snap, it shows why making `--verbose` default works well. Since it doesn't have `contact` at all, the current behaviour from the CLI just shows nothing. If users don't know about --verbose (I didn't for 5 years!) then the CLI gives the impression there's no way to get support.

IMO the only safe solution, given the lack of standardization between snap publishers, is to take the safe route and show everything.